packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts

```diff
@@ -715,20 +725,25 @@ export const reducer: Reducer<State> = (state: State | undefined, action: KnownA
 
 async function onStatusUpdateReceived(
   dispatch: ThunkDispatch<AppState, unknown, KnownAction>,
-  statusUpdate: IStatusUpdate,
+  statusUpdate: IStatusUpdate & {
+    namespace?: string;
+    workspaceId?: string;
+  },
 ) {
-  const { status } = statusUpdate;
+  const { status, namespace, workspaceId } = statusUpdate;
   if (status !== statusUpdate.prevStatus) {
     dispatch({
       type: 'UPDATE_DEVWORKSPACE_STATUS',
       workspaceUID: statusUpdate.workspaceUID,
       message: statusUpdate.message,
       status,
     });
+    if (status === DevWorkspaceStatus.RUNNING && namespace && workspaceId) {
+      injectKubeConfig(namespace, workspaceId).catch(e => console.error(e));
```

Comment:
I'd ask you not to mix async-await and promises chaining if that's possible. In this case, I expected to see the `try-catch` block surrounding the promise and missed the `.catch()` method attached. Such things make it harder to review code now and support it later.

```suggestion
      try {
        injectKubeConfig(namespace, workspaceId);
      } catch (e) {
        console.error(e);
      }
```

Reply:
I don't need an additional delay in this part of the code(injectKubeConfig is an expensive operation). If you have a proposal on how to do it without additional delays, I will be glad to see another solution.

But, as I see, the current solution
```
injectKubeConfig(namespace, workspaceId).catch(e => console.error(e));
```
 is faster. I don't see the reason to add additional delay.

Reply:
I see your point, thanks. Still, I don't think it's a bottleneck that requires us to do this.

Another solution would be not to throw an error in the `injectKubeConfig` method. Will it work for you?

Reply:
thanks!